src/JSPropUpdater.ts

```diff
@@ -55,11 +66,49 @@ export class JSPropUpdater {
     animatedComponent: React.Component<unknown, unknown>
   ) {
     const viewTag = findNodeHandle(animatedComponent);
-    JSPropUpdater._tagToComponentMapping.delete(viewTag);
-    if (JSPropUpdater._tagToComponentMapping.size === 0) {
+    JSPropUpdaterPaper._tagToComponentMapping.delete(viewTag);
+    if (JSPropUpdaterPaper._tagToComponentMapping.size === 0) {
       this._reanimatedEventEmitter.removeAllListeners(
         'onReanimatedPropsChange'
       );
     }
   }
 }
+
+class JSPropUpdaterFabric implements JSPropUpdater {
+  private static _tagToComponentMapping = new Map();
+  private static isInitialized = false;
+
+  constructor() {
+    if (!JSPropUpdaterFabric.isInitialized) {
+      const updater = (viewTag: number, props: unknown) => {
+        const component =
+          JSPropUpdaterFabric._tagToComponentMapping.get(viewTag);
+        component?._updateFromNative(props);
+      };
+      runOnUIImmediately(() => {
+        'worklet';
+        global._jsPropsUpdater = (viewTag: number, props: unknown) => {
+          runOnJS(updater)(viewTag, props);
+        };
+      })();
+      JSPropUpdaterFabric.isInitialized = true;
+    }
+  }
+
+  public addOnJSPropsChangeListener(
+    animatedComponent: React.Component<unknown, unknown>
+  ) {
+    const viewTag = findNodeHandle(animatedComponent);
+    JSPropUpdaterFabric._tagToComponentMapping.set(viewTag, animatedComponent);
+  }
+
+  public removeOnJSPropsChangeListener(
+    animatedComponent: React.Component<unknown, unknown>
+  ) {
+    const viewTag = findNodeHandle(animatedComponent);
+    JSPropUpdaterFabric._tagToComponentMapping.delete(viewTag);
+  }
+}
+
+export default global._IS_FABRIC ? JSPropUpdaterFabric : JSPropUpdaterPaper;
```

Comment:
Do we want to make a fallback for e.g. web here or not necessarily?

Reply:
It seems that it never worked on web 🫣